packages/repo/tests/mst.test.ts

```diff
@@ -277,5 +277,256 @@ describe('utils', () => {
     expect(countPrefixLen('abc', '')).toBe(0)
     expect(countPrefixLen('ab', 'abc')).toBe(2)
     expect(countPrefixLen('abc', 'ab')).toBe(2)
+    expect(countPrefixLen('abcde', 'abc')).toBe(3)
+    expect(countPrefixLen('abc', 'abcde')).toBe(3)
+    expect(countPrefixLen('abcde', 'abc1')).toBe(3)
+    expect(countPrefixLen('abcde', 'abb')).toBe(2)
+    expect(countPrefixLen('abcde', 'qbb')).toBe(0)
+    expect(countPrefixLen('', 'asdf')).toBe(0)
+    expect(countPrefixLen('abc', 'abc\x00')).toBe(3)
+    expect(countPrefixLen('abc\x00', 'abc')).toBe(3)
+  })
+
+  it('counts string (not byte) prefix length', () => {
+    // TODO: these are not cross-language consistent
```

Comment:
hopefully we can wipe all this out soon 😛

Reply:
indeed! was a good dry run for discussion about computing post lengths as well